File: Assets/Scripts/PlayerController.cs
using UnityEngine;
using System.Collections;

public class PlayerController : MonoBehaviour {
	
	Animator playerAnimator { get; set; }
	
	private Rigidbody2D body { get; set; }
	public float jumpForce;
	public float speed;
	
	public float gravity;
	public float drag;
	
	private bool face { get; set; }
	
	// Use this for initialization
	void Start () {
		playerAnimator = GetComponent<Animator> ();
		face = true;
		body = GetComponent<Rigidbody2D>();
		body.gravityScale = gravity;
		body.drag = drag;
	}
	
	// FixedUpdate is called once per frame with physics
	void FixedUpdate () {
		// Movimiento izquierda
		if(Input.GetKey(KeyCode.D)) {
			if(!face) {
				face = true;
				transform.Rotate (new Vector3(0, 180, 0));
			}
			playerAnimator.SetBool("Walking", true);
			body.velocity = new Vector2(speed, body.velocity.y);
		}
		else if(Input.GetKey(KeyCode.A)) {
			if(face) {
				face = false;
				transform.Rotate (new Vector3(0, 180, 0));
			}
			playerAnimator.SetBool("Walking", true);
			body.velocity = new Vector2(-speed, body.velocity.y);
		}
		else {
			playerAnimator.SetBool("Walking", false);
		}
	}
}

File: Assets/Scripts/PlayerController.js
#pragma strict

private var playerAnimator : Animator;
private var body : Rigidbody2D;

var jumpForce : float;
var speed : float;
var gravity : float;
var drag : float;

private var face = true;


function Start () {
	playerAnimator = GetComponent(Animator);
	body = GetComponent(Rigidbody2D);
	body.gravityScale = gravity;
	body.drag = drag;
}

function FixedUpdate () {
	if (Input.GetKey(KeyCode.D)) {
		if(!face) {
			face = true;
			rotarPersonaje();
		}
		playerAnimator.SetBool("Walking", true);
		body.velocity.x = speed;
	}
	else if (Input.GetKey(KeyCode.A)) {
		if(face) {
			face = false;
			rotarPersonaje();
		}
		playerAnimator.SetBool("Walking", true);
		body.velocity.x = -speed;
	}
	else {
		playerAnimator.SetBool("Walking", false);
	}
}

/**
* Funcion que rota el personaje
*/
function rotarPersonaje () {
	transform.Rotate(Vector3(0, 180, 0));
}